### DISCUSSION TITLE
Which version of Java agent to use for productive use cases ?

### DISCUSSION THREAD

**Discussion by santoshkashyap** [open-telemetry/opentelemetry-java-instrumentation#6939] [Q&A] [answered]
Hi Team,

I would like to instrument java applications and ship traces to a tracing backend from CloudFoundry and K8S runtimes. 
I tried the latest JavaAgent 1.19.1. Below is a sample span visualisation in a tracing backend service for spring mvc:

![image](https://user-images.githubusercontent.com/16810482/197177028-34bc0824-3d4d-45de-8e47-6b9e4bb24d04.png)

Similarly, for JDBC and tomcat it has "alpha" suffix. Also verified the previous version 1.18.x. It too seems to have alpha quality for framework support.

Now, randomly tried out older versions of java agent like `1.11.1` and `1.12.0`. These do not have alpha suffix.

![image](https://user-images.githubusercontent.com/16810482/197176732-f0f15285-a439-473c-9e85-50f56689636c.png)

I have also seen the notice on the instrumentation [repository](https://github.com/open-telemetry/opentelemetry-java-instrumentation/releases/tag/v1.19.0) :
> "Note that all artifacts other than io.opentelemetry.javaagent:opentelemetry-javaagent have the -alpha suffix attached to their 
> version number, reflecting that they are still alpha quality and will continue to have breaking changes........"

My question is which version of Java agent is recommended for productive use cases ? Any recommendations for a stable version(slightly older is also fine)

**Best Regards,**
Santosh

**Comment by mateuszrzeszutek** [comment, marked as answer]
Hey @santoshkashyap 

> My question is which version of Java agent is recommended for productive use cases ? Any recommendations for a stable version(slightly older is also fine)

We recommend using the latest version of the javaagent. The agent itself is considered stable; the instrumentations that are included in it are not. This pretty much means that we are allowed to change the telemetry data that is being emitted by the agent, e.g. span names or attributes. This is caused by the fact that the [OTel semantic conventions](https://github.com/open-telemetry/opentelemetry-specification/tree/main/specification/trace/semantic_conventions) that we're based on are not stable themselves, so until the conventions outlined in the spec stabilize we cannot stabilize the instrumentations.

> Now, randomly tried out older versions of java agent like `1.11.1` and `1.12.0`. These do not have alpha suffix.

That is actually an unfortunate bug, all of the instrumentations contained in the previous javaagent versions were still unstable, we were just emitting the javaagent version as `instrumentationLibrary.version`, not the instrumentation version itself.
